diff --git a/helper/builtinplugins/registry.go b/helper/builtinplugins/registry.go
index 325658654..8abb862a4 100644
--- a/helper/builtinplugins/registry.go
+++ b/helper/builtinplugins/registry.go
@@ -55,6 +55,10 @@ import (
 	dbRedshift "github.com/hashicorp/vault/plugins/database/redshift"
 	"github.com/hashicorp/vault/sdk/helper/consts"
 	"github.com/hashicorp/vault/sdk/logical"
+
+	flantGitOps "github.com/flant/negentropy/vault-plugins/flant_gitops"
+	flantIam "github.com/flant/negentropy/vault-plugins/flant_iam/backend"
+	flantIamAuth "github.com/flant/negentropy/vault-plugins/flant_iam_auth/backend"
 )

 // Registry is inherently thread-safe because it's immutable.
@@ -70,26 +74,27 @@ type BuiltinFactory func() (interface{}, error)
 func newRegistry() *registry {
 	reg := &registry{
 		credentialBackends: map[string]logical.Factory{
-			"alicloud":   credAliCloud.Factory,
-			"app-id":     credAppId.Factory,
-			"approle":    credAppRole.Factory,
-			"aws":        credAws.Factory,
-			"azure":      credAzure.Factory,
-			"centrify":   credCentrify.Factory,
-			"cert":       credCert.Factory,
-			"cf":         credCF.Factory,
-			"gcp":        credGcp.Factory,
-			"github":     credGitHub.Factory,
-			"jwt":        credJWT.Factory,
-			"kerberos":   credKerb.Factory,
-			"kubernetes": credKube.Factory,
-			"ldap":       credLdap.Factory,
-			"oci":        credOCI.Factory,
-			"oidc":       credJWT.Factory,
-			"okta":       credOkta.Factory,
-			"pcf":        credCF.Factory, // Deprecated.
-			"radius":     credRadius.Factory,
-			"userpass":   credUserpass.Factory,
+			"alicloud":       credAliCloud.Factory,
+			"app-id":         credAppId.Factory,
+			"approle":        credAppRole.Factory,
+			"aws":            credAws.Factory,
+			"azure":          credAzure.Factory,
+			"centrify":       credCentrify.Factory,
+			"cert":           credCert.Factory,
+			"cf":             credCF.Factory,
+			"gcp":            credGcp.Factory,
+			"github":         credGitHub.Factory,
+			"jwt":            credJWT.Factory,
+			"kerberos":       credKerb.Factory,
+			"kubernetes":     credKube.Factory,
+			"ldap":           credLdap.Factory,
+			"oci":            credOCI.Factory,
+			"oidc":           credJWT.Factory,
+			"okta":           credOkta.Factory,
+			"pcf":            credCF.Factory, // Deprecated.
+			"radius":         credRadius.Factory,
+			"userpass":       credUserpass.Factory,
+			"flant_iam_auth": flantIamAuth.Factory,
 		},
 		databasePlugins: map[string]BuiltinFactory{
 			// These four plugins all use the same mysql implementation but with
@@ -134,6 +139,8 @@ func newRegistry() *registry {
 			"terraform":    logicalTerraform.Factory,
 			"totp":         logicalTotp.Factory,
 			"transit":      logicalTransit.Factory,
+			"flant_iam":    flantIam.Factory,
+			"flant_gitops": flantGitOps.Factory,
 		},
 	}

